### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by daurnimator**
This PR adds a dockerfile so that userdb can be installed using a kubernetes job.

### CODE REVIEW

**Comment by drGrove** [comment]
Should we get ignoring any files? I'd assume we'd want to ignore .git at the minimum

**Comment by daurnimator** [comment]
> Should we get ignoring any files? I'd assume we'd want to ignore .git at the minimum

Good idea; should add a .dockerignore file in this PR.

**Comment by drGrove** [comment]
@daurnimator please add a `.dockerignore` to this PR.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM python:alpine AS gen_schemas
+RUN pip install --no-cache-dir PyYAML
+ADD . /userdb
+WORKDIR  /userdb
+RUN /userdb/json-schemas.py
+
+FROM alpine:latest
+RUN apk --no-cache add make postgresql-client
```
Comment (drGrove) [inline Dockerfile:8]: After installing create a new user and drop into that user.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ADD . /userdb
```
Comment (drGrove) [inline Dockerfile:9]: Change this to:
```
COPY --chown=<user>:<user> . /home/<user>/userdb
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+COPY --from=gen_schemas /userdb/json-schemas.sql.tmp /userdb/json-schemas.sql.tmp
```
Comment (drGrove) [inline Dockerfile:10]: Update path as per above

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+WORKDIR  /userdb
```
Comment (drGrove) [inline Dockerfile:11]: When you drop into the user, you can set the workdir there. Moving this unchanging layer higher up

**Review by drGrove — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (drGrove) [inline Dockerfile:8]: Pin version so multiple builds can be verified

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
+FROM python:alpine AS gen_schemas
+RUN pip install --no-cache-dir PyYAML
+ADD . /userdb
+WORKDIR  /userdb
+RUN /userdb/json-schemas.py
+
+FROM alpine:latest
```
Comment (drGrove) [inline Dockerfile:7]: Avoid using latest.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,12 @@
+FROM python:alpine AS gen_schemas
+RUN pip install --no-cache-dir PyYAML
```
Comment (drGrove) [inline Dockerfile:2]: Pin version of PyYAML

**Comment by lrvick** [comment]
this pr is deprecated by my lrvick-refactor branch which was based on these commits.

**Pull request closed without merging**